hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdb/util/ManagedSstFileReader.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ozone.rocksdb.util;
+
+import org.rocksdb.Options;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.SstFileReader;
+import org.rocksdb.SstFileReaderIterator;
+
+import java.io.Closeable;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * ManagedSSTFileReader provides an abstraction layer using which we can
+ * iterate over multiple underlying SST files transparently.
+ */
+public class ManagedSstFileReader {
+
+  private final Collection<String> sstFiles;
+
+  public ManagedSstFileReader(final Collection<String> sstFiles) {
+    this.sstFiles = sstFiles;
+  }
+  public Stream<String> getKeyStream() throws RocksDBException {
+    final OmSstFileIterator iterator = new OmSstFileIterator(sstFiles);
+    final Spliterator<String> spliterator = Spliterators
+        .spliteratorUnknownSize(iterator, 0);
+    return StreamSupport.stream(spliterator, false).onClose(iterator::close);
+  }
+
+  private static final class OmSstFileIterator implements
+      Iterator<String>, Closeable {
+
+    private final Iterator<String> fileNameIterator;
+    private final Options options;
+    private final ReadOptions readOptions;
+    private String currentFile;
+    private SstFileReader currentFileReader;
+    private SstFileReaderIterator currentFileIterator;
+
+    private OmSstFileIterator(Collection<String> files)
+        throws RocksDBException {
+      // TODO: Check if default Options and ReadOptions is enough.
+      this.options = new Options();
+      this.readOptions = new ReadOptions();
```

Comment:
just for myself learning~ what's difference between Options & ReadOptions?

Reply:
Options is to control the behavior of the _database_.
ReadOptions is to control the behavior of the _get_ method call.